relay-server/src/actors/global_config.rs

```diff
@@ -182,12 +214,10 @@ impl GlobalConfigService {
     ///
     /// We check if we have credentials before sending,
     /// otherwise we would log an [`UpstreamRequestError::NoCredentials`] error.
-    fn update_global_config(&mut self) {
-        self.fetch_handle.reset();
-
-        let upstream_relay = self.upstream.clone();
-        let internal_tx = self.internal_tx.clone();
-
+    fn request_global_config(
+        upstream_relay: Addr<UpstreamRelay>,
+        internal_tx: mpsc::Sender<UpstreamQueryResult>,
+    ) {
```

Comment:
imo, this name makes more sense, as the previous name implies that this function is the one that updates the global config.  
 
I changed it to an associated function so that I could call it from `fn new` easily.  The fetch_handle.reset() is done in spawn_handler now.